std/algorithm/mutation.d

```diff
@@ -365,7 +365,7 @@ Returns:
     The unfilled part of target
  */
 TargetRange copy(SourceRange, TargetRange)(SourceRange source, TargetRange target)
-if (isInputRange!SourceRange && isOutputRange!(TargetRange, ElementType!SourceRange))
+if ((isInputRange!SourceRange || isArray!SourceRange) && isOutputRange!(TargetRange, ElementType!SourceRange))
```

Comment:
Most functions in phobos work with ranges, but not with static arrays. Array slices (dynamic arrays) are already ranges and so are covered by the `isInputRange` constraint. If the consensus is that we want to allow `copy` to receive static array arguments, I suggest being explicit and change `isArray` to `isStaticArray`.

Additionally, receiving large objects by value can be a performance issue, so I suggest changing the `SourceRange source` parameter to `auto ref SourceRange source` to avoid unnecessary copying large static arrays.

Reply:
@ZombineDev, Thank you! It has been adjusted to your suggestion: a5b83ca